### ISSUE TITLE
change ChainerConverter's strategy

### ISSUE THREAD

**Issue by Kiikurage** [mil-tokyo/webdnn#419]
## Problem

Currently, `ChainerConverter`'s conversion strategy is:

1. Convert all chainer's parameter variable into `webdnn.ConstantVariable`
    1. Assume that all named variables are parameter variable, and all variables without name is not parameter variable
    2. Each parameter variable's data order is determined based on chainer's default data order.
2. Convert each `Function` object and its intermediate output variables from input to output.

However this assumption is not good because there are some irregular case such as `BatchNormalizationFunction`, `Deconvolution2D`.

## Proposal

Convert all variables not only output but also input in each converter handler (`KerasConverter` is designed based on this strategy).
In this case, irregular function can be handled by corresponding converter handler.

**Comment by Kiikurage** [comment]
done #429
